test/e2e/proxy_test.go

```diff
@@ -150,16 +178,103 @@ func TestProxyFlow(t *testing.T) {
+				err := proxyCl.List(context.TODO(), applicationList, &client.ListOptions{Namespace: tenantNsName(user.compliantUsername)})
+				// User should be able to list applications
+				require.NoError(t, err)
+				assert.NotNil(t, applicationList)
+				assert.NotEmpty(t, applicationList)
+
+				// Double check that the applicationList using a regular client (non-proxy)
+				applicationListWS := &appstudiov1.ApplicationList{}
+				err = user.expectedMemberCluster.Client.List(context.TODO(), applicationListWS, &client.ListOptions{Namespace: tenantNsName(user.compliantUsername)})
+				require.NoError(t, err)
+				require.NotEmpty(t, applicationListWS)
+				assert.Equal(t, applicationListWS.Items, applicationList.Items)
+			})
+
+			t.Run("use proxy to patch a HAS Application CR in the user appstudio namespace via proxy API and use websocket to watch it patched", func(t *testing.T) {
+				// Patch application
+				applicationName := user.getApplicationName(1)
+				patchString := "Patched application for proxy test"
+				// Get application
+				proxyApp := user.getApplication(t, proxyCl, applicationName)
+				// Patch for DisplayName
+				patchPayload := []patchStringValue{{
+					Op:    "replace",
+					Path:  "/spec/displayName",
+					Value: patchString,
+				}}
+				patchPayloadBytes, _ := json.Marshal(patchPayload)
+
+				// Appply Patch
+				err := proxyCl.Patch(context.TODO(), proxyApp, client.RawPatch(types.JSONPatchType, patchPayloadBytes))
+				require.NoError(t, err)
+
+				// Get patched app and verify patched DisplayName
+				patchedApp := user.getApplication(t, proxyCl, applicationName)
+				assert.Equal(t, patchedApp.Spec.DisplayName, patchString)
+
+				// Double check that the Application is patched using a regular client (non-proxy)
+				wsApp := user.getApplicationWithoutProxy(t, applicationName)
+				assert.Equal(t, wsApp.Spec.DisplayName, patchString)
+			})
+
+			t.Run("use proxy to delete a HAS Application CR in the user appstudio namespace via proxy API and use websocket to watch it deleted", func(t *testing.T) {
+				// Delete applications
+				for i := 0; i < 2; i++ {
+					applicationName := user.getApplicationName(i)
+					// Get application
+					proxyApp := user.getApplication(t, proxyCl, applicationName)
+					// Delete
+					err := proxyCl.Delete(context.TODO(), proxyApp)
+					require.NoError(t, err)
+					err = w.WaitForApplicationDeletion(
+						user.expectedMemberCluster.RetryInterval,
+						user.expectedMemberCluster.Timeout,
+						applicationName,
+					)
+					require.NoError(t, err)
+
+					// Double check that the Application is deleted using a regular client (non-proxy)
+					namespacedName := types.NamespacedName{Namespace: tenantNsName(user.compliantUsername), Name: applicationName}
+					originalApp := &appstudiov1.Application{}
+					err = user.expectedMemberCluster.Client.Get(context.TODO(), namespacedName, originalApp)
+					require.Error(t, err) //not found
```

Comment:
Check the error to make sure it's actually 404.

Reply:
Added check for k8serr.IsNotFound(err).